### PR TITLE
Extract collection [3]

### PR DESCRIPTION
**Pull request by JojiiOfficial**
### All Submissions:

* [ ] Contributions should target the `dev` branch. Did you create your branch from `dev`?
* [ ] Have you followed the guidelines in our Contributing document?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [ ] Does your submission pass tests?
2. [ ] Have you formatted your code locally using `cargo +nightly fmt --all` command prior to submission?
3. [ ] Have you checked your code using `cargo clippy --all --all-features` command?

### Changes to Core Features:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your core changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: src/client/collection.rs [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+                        request: Some(DeleteShardKey {
+                            shard_key: Some(ShardKey::from(shard_key.clone())),
+                        }),
+                        timeout: None,
+                    })
+                    .await?;
+                Ok(result.into_inner())
+            })
+            .await?)
+    }
```
Comment (timvisee) [inline src/client/collection.rs:333]: We might want to move this into a `sharding_keys` module, similar to this: https://github.com/qdrant/qdrant/blob/master/lib/collection/src/collection/sharding_keys.rs

Comment (generall) [reply src/client/collection.rs:333]: It can be a next step

**Review by generall — APPROVED** [review]

**Comment by generall** [comment]
@JojiiOfficial pls resolve conflict and feel free to merge

File: src/client/collection.rs (hunk 1 above)
Comment (timvisee) [reply src/client/collection.rs:333]: Yeah, we discussed to do this at the end to not mess up the following PRs that are already queued.

**Review by timvisee — APPROVED** [review]

**Pull request merged**